src/Widgets/DiskBar.vala

```diff
@@ -0,0 +1,85 @@
+// -*- Mode: vala; indent-tabs-mode: nil; tab-width: 4 -*-
+/*-
+ * Copyright (c) 2016-2018 elementary LLC. (https://elementary.io)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: Michael Aaron Murphy <michael@system76.com>
+ */
+
+public class Installer.DiskBar: Gtk.Box {
+    public string disk_name;
+    public string disk_path;
+    public uint64 size;
+
+    public Gtk.Box label;
+
+    public DiskBar (
+        string model,
+        string path,
+        uint64 size,
+        GLib.Array<PartitionBar> partitions
+    ) {
+        this.disk_name = model;
+        this.disk_path = path;
+        this.size = size;
+
+        generate_label ();
+        generate_bar (partitions);
+    }
+
+    construct {
+        this.orientation = Gtk.Orientation.HORIZONTAL;
+        this.hexpand = true;
+        this.get_style_context ().add_class("disk-bar");
```

Comment:
I think instead of recreating this look with custom classes and styling, we could lean a little more on the existing `.storage-bar` styles that are already implemented in the elementary and Pop! stylesheets. The container box would need `.storage-bar`, the disk would need `.trough`, and then each partition inside would need `.fill-block` and then could have a custom class for coloring if we want.

This would set borders, shadows, padding, etc. to what the stylesheet already uses to represent a disk and its contents elsewhere.

Reply:
By container box, are you referring to the  disk bar + disk label?